### PR TITLE
Chore: Add some e2e tests for repeating behaviour

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

For #42512

**Special notes for your reviewer**:

### CODE REVIEW

**Review by hugohaggmark — APPROVED** [review]
Hurray for the Mob 🔱

**Comment by sdboyer** [comment]
I'm on vacation this week and next.

If this is what I think it is (we didn't update the schema when we made data sources into objects), it's going to take a bit of doing to get it done. If this is blocking something critical, I can see if I can find some time before the new year.

**Comment by ashharrison90** [comment]
@sdboyer no worries, it's not critical. 

_some_ errors are related to the data source object thing. but there are others that are unrelated (null values, it not recognising `timeseries`/`repeats`/`collapsed`). 🤷 

enjoy your holidays! 🎅

**Comment by hugohaggmark** [comment]
Yeah enjoy your holidays @sdboyer, sorry for ping.

**Comment by sdboyer** [comment]
no worries, I'll get on it when I get back then.

def concerning that stripnulls mangled it - if we've found a case where we're expecting explicit null values, we'll have more to unravel.

but, later!

**Comment by ashharrison90** [comment]
@sdboyer lemme know if you wanna pair up on looking into this 🙂

File: scripts/stripnulls.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@
 SED=$(command -v gsed)
 SED=${SED:-"sed"}
 
-FILES=$(grep -rl '"schemaVersion": 3[0123]' devenv)
+FILES=$(grep -rl '"schemaVersion": 3[01234]' devenv)
```
Comment (sdboyer) [inline scripts/stripnulls.sh:10]:
```suggestion
FILES=$(grep -rl '"schemaVersion": 3[34]' devenv)
```

Comment (sdboyer) [reply scripts/stripnulls.sh:10]: With the changes to the schema, we're not actually supporting those older versions anymore, so might as well narrow the margin for error with the hacky shell script.

**Review by sdboyer — CHANGES_REQUESTED** [review]
Wonderful! thanks so much for pairing on this, and for updating the other dashboards we'd previously been testing.

Only one thing - we do need the changes to the Go test you made to only test `schemaVersion` > 32. It's unlikely we end up creating any in the future, but there's no sense in leaving ourselves a timebomb in case it does happen, or otherwise leaving that test in a state that's divergent from reality.

**Review by sdboyer — APPROVED** [review]
looks good, thank you!

**Comment by ashharrison90** [comment]
for future reference: the cue schema changes were fixed with a combination of running the tests using `go test -run ^TestDevenvDashboardValidity$ github.com/grafana/grafana/pkg/schema/load -count=1 -v`, and the `stripnulls` script

**Pull request merged**